lib/perl/Genome/Disk/Detail/Allocation/Unarchiver.pm

```diff
@@ -177,4 +133,49 @@ sub _swap_shadow_allocation {
     return 1;
 }
 
+sub _do_unarchive_cmd {
+    my $class = shift;
+    my $id = shift;
+    my $cmd = shift;
+
+    # It's very possible that if no commit is on, the volumes/allocations
+    # being dealt with are test objects that don't exist out of this local
+    # UR context, so bsubbing jobs would fail.
+    if ($ENV{UR_DBI_NO_COMMIT}) {
+        Genome::Sys->shellcmd(cmd => $cmd);
+    } else {
+        # If this process should be killed, the LSF job needs to be cleaned up
+        my ($job_id, $status);
+
+        # Signal handlers are added like this so an anonymous sub can be
+        # used, which handles variables defined in an outer scope
+        # differently than named subs (in this case, $job_id).
+        my $sig_handler = sub {
+            print STDERR "Cleanup activated within allocation, ",
+                "cleaning up LSF jobs\n";
+            eval { Genome::Sys->kill_lsf_job($job_id) } if $job_id;
+            die "Received signal, exiting.";
+        };
+        
+        local @SIG{'INT','TERM'} = ($sig_handler, $sig_handler);
+
+        # Entire path must be wrapped in quotes because older allocation
+        # IDs contain spaces If the command isn't wrapped in quotes, the
+        # '&&' is misinterpreted by bash (rather than being "bsub '1 && 2'
+        # it is looked at as 'bsub 1' && '2')
+        ($job_id, $status) = Genome::Sys->bsub_and_wait(
+            queue => Genome::Config::get('archive_lsf_queue'),
+            job_group => '/unarchive',
+            log_file => "/tmp/$id",
+            cmd => $cmd,
```

Comment:
If you pass the command as an arrayref, then you don't have to worry about special characters messing up the shell
```cmd => ['tar', '-C', $target_path, '-xf', $tar_path]```